### PR TITLE
docs: Amend metrics doc + diagrams

### PR DESCRIPTION
**Pull request by NHingerl**
## Description

Changes proposed in this pull request:

- combine step 1a+b and use toggle
- minor changes in metrics doc
- update metrics and traces diagram to match our guidelines

## Related Issues and Documents

Closes:

Related issues:

follow-up to https://github.com/kyma-project/telemetry-manager/pull/364

### CODE REVIEW

**Review by skhalash — DISMISSED** [review]

File: docs/user/04-metrics.md [hunk 1]
```diff
@@ -329,7 +342,9 @@ spec:
 The agent will configure the [kubletstatsreceiver](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/kubeletstatsreceiver) for the metric groups `pod` and `container`. With that, [system metrics](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/receiver/kubeletstatsreceiver/documentation.md) related to containers and pods will get collected.
```
Comment (grego952) [inline docs/user/04-metrics.md:342]:
```suggestion
The agent configures the [kubletstatsreceiver](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/kubeletstatsreceiver) for the metric groups `pod` and `container`. With that, [system metrics](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/receiver/kubeletstatsreceiver/documentation.md) related to containers and pods get collected.
```

File: docs/user/04-metrics.md [hunk 2]
```diff
@@ -368,6 +385,7 @@ You activated a MetricPipeline and metrics start streaming to your backend. To v
 ## Limitations
 
 The metric gateway setup is based on the following assumptions:
+
 - The collector has no autoscaling options and has a limited resource setup of 1 CPU and 1 GiB memory.
 - Batching is enabled, and a batch will contain up to 512 metrics/batch.
```
Comment (grego952) [inline docs/user/04-metrics.md:390]:
```suggestion
- Batching is enabled, and a batch contains up to 512 metrics/batch.
```

File: docs/user/04-metrics.md [hunk 3]
```diff
@@ -56,38 +58,47 @@ Furthermore, the manager takes care of the full lifecycle of the Gateway Deploym
...
+      spec:
+        output:
+          otlp:
+            endpoint:
+              value: https://backend.example.com:4317
+      ```
+  </details>
+  <details>
+    <summary>HTTP</summary>
+    To use the HTTP protocol, use the `protocol` attribute:
```
Comment (IwonaLanger) [inline docs/user/04-metrics.md:86]:
```suggestion
    To use the HTTP protocol, use the <code>protocol</code> attribute:
```

Comment (IwonaLanger) [inline docs/user/04-metrics.md:86]: HTML and Markdown do not work together.

File: docs/user/04-metrics.md [hunk 4]
```diff
@@ -376,50 +394,64 @@ The metric gateway setup is based on the following assumptions:
...
+
+Remedy: You can either set up 'STRICT' mTLS mode or HTTP scraping:
+<div tabs>
+  <details>
+    <summary>Strict mTLS</summary>
+    Configure the workload using 'STRICT' mTLS mode (for example, by applying a corresponding PeerAuthentication).
+  </details>
+  <details>
+    <summary>HTTP scraping</summary>
+    Set up scraping through HTTP by applying the `prometheus.io/scheme=http` annotation.
```
Comment (IwonaLanger) [inline docs/user/04-metrics.md:455]:
```suggestion
    Set up scraping through HTTP by applying the <code>prometheus.io/scheme=http</code> annotation.
```

File: docs/user/04-metrics.md [hunk 5]
```diff
@@ -33,7 +32,9 @@ Optionally, the Telemetry module provides a DaemonSet of an [OTel Collector](htt
 1. The metric data is consumed using the backend system.
 
 ### Metric Gateway
+
 In a Kyma cluster, the metric gateway is the central component to which all components can send their individual metrics. The gateway collects, enriches, and dispatches the data to the configured backend. The gateway is based on the [OTel Collector](https://opentelemetry.io/docs/collector/) and comes with a concept of pipelines consisting of receivers, processors, and exporters, with which you can flexibly plug pipelines together (see [Configuration](https://opentelemetry.io/docs/collector/configuration/). Kyma's MetricPipeline provides a hardened setup of an OTel Collector and also abstracts the underlying pipeline concept. Such abstraction has the following benefits:
+
```
[the remark this answers is not in the document]
Comment (NHingerl) [reply docs/user/04-metrics.md:37]: I opted for full stops. Thanks for spotting it!

File: docs/user/04-metrics.md [hunk 6]
```diff
@@ -42,6 +43,7 @@ In a Kyma cluster, the metric gateway is the central component to which all comp
 The downside is that only a limited set of features is available. If you want to avoid this downside, bring your own collector setup. The current feature set focuses on providing the full configurability of backends integrated by OTLP.
 
 ### Metric Agent
+
 If a MetricPipeline configures a feature in the `input.application` section, an additional DaemonSet is deployed acting as an agent. The agent is also based on an [OTel Collector](https://opentelemetry.io/docs/collector/) and encompasses the collection and conversion of Prometheus-based metrics. Hereby, the workload puts an `prometheus.io/scrape` annotation on the specification of the Pod or service, and the agent collects it. The agent pushes all data in OTLP to the central gateway.
 
 ### Telemetry Manager
```
[the remark this answers is not in the document]
Comment (NHingerl) [reply docs/user/04-metrics.md:49]: Oh lord, the whole diagram is weirdly wonky. Thanks for spotting this.

File: docs/user/04-metrics.md [hunk 7]
```diff
@@ -56,38 +58,47 @@ Furthermore, the manager takes care of the full lifecycle of the Gateway Deploym
...
+      name: backend
+    spec:
+      output:
+        otlp:
+          protocol: http
+          endpoint:
+            value: https://backend.example.com:4318
+    ```
+  </details>
+</div>
```
[the remark this answers is not in the document]
Comment (NHingerl) [reply docs/user/04-metrics.md]: I got the number of blank spaces wrong. Fixed it now :)

**Review by IwonaLanger — APPROVED** [review]

**Pull request merged**